js/index.js

```diff
@@ -1354,8 +1357,10 @@ function DataManager() {
 			});
 		} else {
 			oTable.fnClearTable();
-			oTable.fnAddData(dataList);
-			oTable.fnAdjustColumnSizing();
+			if (dataList.length > 0){
+				oTable.fnAddData(dataList);
+				oTable.fnAdjustColumnSizing();
+			}
```

Comment:
As you say, when there is no data, all renderers are called with null/[]. I found this:

https://datatables.net/forums/discussion/21423/empty-json-array-passed-to-aocolumns-causing-issues-after-1-9-to-1-10-migration

Your workaround is simple, let's keep it.